docs/diagnostics/educational-notes/nominal-types

```diff
@@ -0,0 +1,7 @@
+Nominal types
+-------------
```

Comment:
This looks like Markdown. Is it? If so, it should probably have a .md extension; if not, we should think about whether we want these files formatted in some standard way so that they can have a rich graphical display.

Reply:
I think we'll inevitably want to use Markdown here, we may as well commit to it now and add the extensions. We already have a CommonMark dependency so I think the plan is to keep markup minimal until we can figure out a decent terminal rendering solution